paddle/fluid/operators/gather_op_npu.cc

```diff
@@ -0,0 +1,117 @@
+/* Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <memory>
+#include <string>
+#include <vector>
+#include "paddle/fluid/operators/npu_op_runner.h"
+#include "paddle/fluid/operators/gather_op.h"
+#include "paddle/fluid/framework/tensor_util.h"
+
+namespace paddle {
+namespace operators {
+
+inline framework::Tensor UnsqueezeTo(const framework::Tensor& src, int ndims) {
+  const framework::DDim& shape = src.dims();
+  int rank = shape.size();
+  framework::Tensor res;
+  res.ShareDataWith(src);
+  PADDLE_ENFORCE_LE(
+      rank, ndims,
+      platform::errors::InvalidArgument(
+          "The input Tensor's rank should be less than or equal to ndims"
+          "Received input Tensor's rank = %d, ndims = %d",
+          rank, ndims));
+  if (rank < ndims) {
+    std::vector<int64_t> new_dim(ndims, 1);
+    for (int i = ndims - rank; i < ndims; i++) {
+      new_dim[i] = shape[i - ndims + rank];
+    }
+    res.Resize(framework::make_ddim(new_dim));
+  }
+  return res;
+}
```

Comment:
It is duplicated with UnsqueezeTo in kron.h? If yes, we can just include kron.h.

Reply:
> It is duplicated with UnsqueezeTo in kron.h? If yes, we can just include kron.h.

done